src/rpc/rawtransaction.cpp

```diff
@@ -90,6 +90,15 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
                     txinwitness.push_back(HexStr(item.begin(), item.end()));
                 }
                 in.push_back(Pair("txinwitness", txinwitness));
+                // try to parse the last item as a pubkey
+                // if someone constructed the script to look like a pubkey, then he's insane
```

Comment:
nit: suggest you add a bit more context to this comment. eg:

In version 0, there are two valid witness types: P2WPKH and P2WSH. P2WPKH witnesses have two items on the witness stack, the second of which is a public key. We test whether the last item on the witness stack is a valid public key, and if it isn't we try to parse it as a redeem script.